source/core/interface_tools.cc

```diff
@@ -286,6 +334,890 @@ InterfaceTools::SignedDistanceSolver<dim, VectorType>::get_signed_distance()
+
+  /* Compute targeted global volume. It corresponds to the one enclosed by
+  the level 0 of the level_set vector (same volume as the one enclosed
+  by iso-contour 0.5 of the phase fraction).*/
+  const double global_volume = compute_volume(
+    *mapping, dof_handler, *fe, level_set, 0.0, mpi_communicator);
+
+  /* Initialization of values for the secant method. The subscript nm1 (or n
+  minus 1) stands for the previous secant iteration (it = n-1), the
+  subscript n stands for the current iteration and the subscript np1 stands
+  for the next iteration (it = n+1).*/
+  double global_volume_nm1 = 0.0;
+  double global_volume_n   = 0.0;
+
+  double global_delta_volume_nm1 = 0.0;
+  double global_delta_volume_n   = 0.0;
+
+  double global_delta_volume_prime = 0.0;
+
+  /* Global constant C that we are solving for to obtain
+      DeltaV(phi* + C*eta) = V_VOF - V(phi* + C*eta) = 0
+  where eta is the cell-wise correction computed with
+  compute_cell_wise_volume_correction()*/
+  double C_nm1 = 0.0;
+  double C_n   = 0.0;
+  double C_np1 = 0.0;
+
+  /* Compute the volume and the difference with the targeted volume for 1st
+  initial guess of the correction function (xi_nm1 = C_nm1*eta)*/
+  C_nm1 = 1.0;
+  LinearAlgebra::distributed::Vector<double> signed_distance_0(
+    signed_distance_with_ghost);
+  signed_distance_0.add(C_nm1, volume_correction);
+
+  // Update_ghost_values is required for cell-wise volume computations
+  signed_distance_0.update_ghost_values();
+
+  global_volume_nm1 = compute_volume(
+    *mapping, dof_handler, *fe, signed_distance_0, 0.0, mpi_communicator);
+
+  global_delta_volume_nm1 = global_volume - global_volume_nm1;
+
+  // Initialize the 2nd initial guest
+  C_n = 1e-6 * global_volume;
+
+  // Store the initial volume for the stop criterion
+  const double global_volume_0 = global_volume_nm1;
+
+  // Initialize secant method it and update
+  const double tol           = 1e-12;
+  unsigned int secant_it     = 0;
+  unsigned int secant_max_it = 20;
+  double       secant_update = 1.0;
+
+  // Secant method
+  while (abs(secant_update) > tol &&
+         abs(global_delta_volume_nm1) > tol * global_volume_0 &&
+         secant_it < secant_max_it)
+    {
+      secant_it += 1;
+
+      LinearAlgebra::distributed::Vector<double> signed_distance_n(
+        signed_distance_with_ghost);
+      signed_distance_n.add(C_n, volume_correction);
+      signed_distance_n.update_ghost_values();
+
+      global_volume_n = compute_volume(
+        *mapping, dof_handler, *fe, signed_distance_n, 0.0, mpi_communicator);
+
+      global_delta_volume_n = global_volume - global_volume_n;
+
+      global_delta_volume_prime =
+        (global_delta_volume_n - global_delta_volume_nm1) / (C_n - C_nm1 + tol);
+
+      secant_update =
+        -global_delta_volume_n / (global_delta_volume_prime + tol);
+
+      C_np1 = C_n + secant_update;
+      C_nm1 = C_n;
+      C_n   = C_np1;
+
+      global_volume_nm1       = global_volume_n;
```

Comment:
I'm not sure why `global_volume_nm1` is updated, it does not seem that it is used in the `while` loop or after; I am maybe missing something.

```suggestion
```

Reply:
I leave it just for completeness :)